feat: amdvlk

VirtualFreeEx opened the pull request:
Aims to fulfill package request #2070. Discussion (more of a monologue) on some key issues I have with both the proposal itself and my current implementation of it will be provided below in comments.

VirtualFreeEx committed:
feat: DirectXShaderCompiler

VirtualFreeEx committed:
chore(DirectXShaderCompiler): refactor, use subpackages

VirtualFreeEx committed:
fix(DirectXShaderCompiler): proper changelog versions

VirtualFreeEx committed:
fix(DirectXShaderCompiler): proper changelog date

VirtualFreeEx committed:
fix(DirectXShaderCompiler): quiet setup

VirtualFreeEx committed:
Merge branch 'terrapkg:frawhide' into ffi/package/amdvlk

VirtualFreeEx commented:
1. AMDVLK already provides their builds of the driver for RHEL/as an RPM, so is this package _really needed_?
2. DirectXShaderCompiler
    1. (by default) Should we build with external libraries or with libraries from Fedora's repositories?

VirtualFreeEx committed:
feat(DirectXShaderCompiler): external libraries bcond

VirtualFreeEx commented:
> AMDVLK already provides their builds of the driver for RHEL/as an RPM, so is this package _really needed_?

Specifically on this topic: ~~I believe that the upstream's specfile~~ could be used instead if we don't trust the upstream-provided binaries. Alternatively, could AMDVLK be somehow shipped as the upstream-provided package?

Edit 1: Upstream does not provide a specfile and instead uses CPack to create the package.

madonuko commented:
we (fyra) are having an ongoing discussion on whether we allow shipping upstream-provided rpms directly, but before we draw a conclusion I think it's not a bad idea to just provide our own rpms anyway

madonuko commented on anda/lib/directxshadercompiler/DirectXShaderCompiler.spec:
```diff
@@ -0,0 +1,243 @@
+    -DLLVM_BUILD_TOOLS=0 \
+    -DSPIRV_BUILD_TESTS=0 \
+    -DLLVM_ENABLE_WERROR=0 \
+    -DBUILD_SHARED_LIBS=OFF \
```
interesting, I wonder why is this off but we still have the shared libs?

VirtualFreeEx replied:
i.. dont really have an idea to be honest.

madonuko commented on anda/lib/directxshadercompiler/DirectXShaderCompiler.spec:
```diff
@@ -0,0 +1,243 @@
+%check
+%ctest
+ 
+# Files.
+%files tools
+%{_libdir}/%{name}/bin/
+%{_bindir}/dxa
+%{_bindir}/dxc
+%{_bindir}/dxl
+%{_bindir}/dxopt
+%{_bindir}/dxr
+%{_bindir}/dxv
+
```
why putting everything into the subpackage when you can just put them into the main package?

VirtualFreeEx replied:
fair point, subpackage removed in local tree, will be pushed later

madonuko commented on anda/lib/directxshadercompiler/DirectXShaderCompiler.spec:
```diff
@@ -0,0 +1,243 @@
+%{_libdir}/%{name}/lib/libdxil.so
+
+%files static
+%{_libdir}/%{name}/lib/libHLSLTestLib.a
```
could just do `*.a`?

VirtualFreeEx replied:
done

madonuko commented on anda/lib/directxshadercompiler/DirectXShaderCompiler.spec:
```diff
@@ -0,0 +1,243 @@
+    -DBUILD_SHARED_LIBS=OFF \
+    -DLLVM_VERSION_SUFFIX=dxc \
+    -DSPIRV_WARN_EVERYTHING=0 \
+    -DCMAKE_INSTALL_PREFIX="%{_libdir}/%{name}" \
```
why not just `%_prefix`?

GildedRoach replied:
I looked at the files. This should be built into the prefix. This does also include CMake files which are currently in the wrong location meaning CMake probably cannot find them and which also belong in the devel package.

madonuko replied:
this

VirtualFreeEx replied:
because it would possibly conflict with files from llvm's files

GildedRoach replied:
The package is not usable by CMake *or* LLVM this way. This has CMake files and compiler files that must be in the intended locations or they can't be used.

VirtualFreeEx replied:
removed the CMake files (i dont think i was correct in the first place with including them)

i dont think that llvm would use this package by itself since this package happens to be a fork of llvm and specifically clang afaik, not something that uses it at runtime

VirtualFreeEx committed:
Merge branch 'terrapkg:frawhide' into ffi/package/amdvlk

VirtualFreeEx committed:
feat(DirectXShaderCompiler)!: bump, remove some subpackages

BREAKING CHANGE: The `tools` and the `cmake-utils`
subpackages no longer exist.

VirtualFreeEx committed:
chore(DirectXShaderCompiler): use git_setup macro

VirtualFreeEx committed:
chore(DirectXShaderCompiler): auto-update

VirtualFreeEx committed:
feat: amdvlk

VirtualFreeEx committed:
chore(amdvlk): attempt to fix ci build, documentation

korewaChino commented:
AMDVLK is dead, long live RADV

https://www.phoronix.com/news/AMDVLK-Discontinued

GildedRoach commented:
> AMDVLK is dead, long live RADV
> 
> https://www.phoronix.com/news/AMDVLK-Discontinued

While that is true, some video and other editing software do not accept or work with Mesa. It's a frustrating situation.
